### ISSUE TITLE
Suggestions for `update` API improvements

### ISSUE THREAD

**Issue by kress95** [PaackEng/elm-ui-dropdown#45]
I have _buncha_ feedback suggestions from integrating this library with [`paack-ui`](
https://github.com/PaackEng/paack-ui/pull/244), <sub><sup>and also my previous experience integrating Elm [with `deno`'s HTTP](https://gist.github.com/kress95/524645e58cee593c7fc9d2b3e0504f99)</sup></sub>

# `Dropdown` module 

## `Effect` type 

- Remove `Loopback`
- Add `Selected item` (or a better name)

## `InternalConfig` record type

- Remove `onSelectMsg`
  - Remove every propagation of it

Because the new effect already captures all the work `onSelectMsg` was doing.

## `update` function

- Remove `Config item msg model` parameter
- Add `{ onSelected : item -> Cmd msg }` (or a better name) as the first parameter

Because keeping config here requires the update flow to know too much about the view, the `onSelected` function also allows avoiding the loopback.

## `updateWithoutPerform` function

- Remove `Config item msg model` parameter

And nothing more is needed here.

---

Let's bring this to **pristine** state 💪

**Comment by PedroHLC** [comment]
> Remove Loopback; Add Selected item (or a better name)

I disagree, it's not very clear how one should implement that effect (or their simulation) on their own...

> Remove Config item msg model parameter

oh God, please do this!

**Comment by kress95** [comment]
> I disagree, it's not very clear how one should implement that effect (or their simulation) on their own...

It doesn't matter "how one should implement that effect", we should not expect to have a canonical implementation for every effect. For this one, the outcome is up to the user.

We just created a lot of effects with canonical implementations elsewhere because it's convenient for their purpose (E2E testing).

---

> Remove Config item msg model parameter

You'll have to remove loopback to do that.
